Assets/MRTK/Core/Utilities/XRSettingsUtilities.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#if UNITY_2019_3_OR_NEWER
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.XR;
+#endif // UNITY_2019_3_OR_NEWER
+
+namespace Microsoft.MixedReality.Toolkit.Utilities
+{
+    /// <summary>
+    /// Utilities that abstract XR settings functionality so that the MRTK need not know which
+    /// implementation is being used.
+    /// </summary>
+    public static class XRSettingsUtilities
+    {
+#if UNITY_2019_3_OR_NEWER
+        private static bool? isLegacyXRActive = null;
+#endif // UNITY_2019_3_OR_NEWER
+
+        /// <summary>
+        /// Checks if an XR SDK plug-in is installed that disables legacy VR. Returns false if so.
+        /// </summary>
+        public static bool IsLegacyXRActive
+        {
+            get
+            {
+#if UNITY_2019_3_OR_NEWER
+                if (!isLegacyXRActive.HasValue)
+                {
+                    isLegacyXRActive = true;
+
+                    List<XRDisplaySubsystemDescriptor> descriptors = new List<XRDisplaySubsystemDescriptor>();
+                    SubsystemManager.GetSubsystemDescriptors(descriptors);
```

Comment:
@davidkline-ms @MaxWang-MS @RogPodge @CDiaz-MS I've found that the AppData folder on-device doesn't get cleaned up on an uninstall, so sometimes the subsystem lists are cached long term. For example from the logs (even though I'm running legacy XR on Unity 2019):

```
[Subsystems] Discovering subsystems at path C:/Data/Users/DefaultAccount/AppData/Local/DevelopmentFiles/Microsoft.MixedReality.ToolkitVS.Master_ARM64.kurti/Data/UnitySubsystems
[Subsystems] No descriptors matched for  examples in UnitySubsystems/MicrosoftOpenXRPlugin/UnitySubsystemsManifest.json.
[Subsystems] 1 'inputs' descriptors matched in UnitySubsystems/MicrosoftOpenXRPlugin/UnitySubsystemsManifest.json
[Subsystems] No descriptors matched for  displays in UnitySubsystems/MicrosoftOpenXRPlugin/UnitySubsystemsManifest.json.
[Subsystems] 1 'meshings' descriptors matched in UnitySubsystems/MicrosoftOpenXRPlugin/UnitySubsystemsManifest.json
[Subsystems] No descriptors matched for  examples in UnitySubsystems/UnityOpenXR/UnitySubsystemsManifest.json.
[Subsystems] 1 'inputs' descriptors matched in UnitySubsystems/UnityOpenXR/UnitySubsystemsManifest.json
[Subsystems] 1 'displays' descriptors matched in UnitySubsystems/UnityOpenXR/UnitySubsystemsManifest.json
[Subsystems] No descriptors matched for  meshings in UnitySubsystems/UnityOpenXR/UnitySubsystemsManifest.json.
[Subsystems] No descriptors matched for  examples in UnitySubsystems/WindowsMRXRSDK/UnitySubsystemsManifest.json.
[Subsystems] 1 'inputs' descriptors matched in UnitySubsystems/WindowsMRXRSDK/UnitySubsystemsManifest.json
[Subsystems] 1 'displays' descriptors matched in UnitySubsystems/WindowsMRXRSDK/UnitySubsystemsManifest.json
[Subsystems] 1 'meshings' descriptors matched in UnitySubsystems/WindowsMRXRSDK/UnitySubsystemsManifest.json
```

The only way I've found to properly clear the AppData folder is to set "Remove Non-Layout Files from Device" under the Remote Machine (not Device, unfortunately) Debugging settings in VS:

![image](https://user-images.githubusercontent.com/3580640/115329925-e57f3080-a147-11eb-9ee7-98d993ced851.png)

This will only affect people who installed a version of an app that supported XR SDK and has since rolled back to legacy (which I'd assume would be 100% developers only).

How do we want to approach this? Is this "edge case" enough, with a solid workaround, that we're okay with it? I'm not sure another way to check this info, and it works perfectly fine in-editor. It's just on-device.

Reply:
I think the use case of going from xrsdk to legacy should be uncommon enough that this shouldn't block the PR from getting in, though we might want to have a section in our documentation which explicitly lays out this step for developers who do so.

Reply:
I agree with @RogPodge. As long as we document this clearly, I think this PR should be fine.

The one thing I would be concerned with is the behavior for non-developer customers (ex: if they uninstall an XR SDK build of an app and attempt to install a legacy, older?, build).

Reply:
Also, do these messages break the app's behavior? Or just clutter the logs?

Reply:
Break the app's behavior. The app reports that it's running XR SDK (because stale subsystem json files are found), but Unity is actually running legacy. Thus, the providers are reading from the wrong APIs and no input comes through.

Reply:
> Thus, the providers are reading from the wrong APIs and no input comes through.

Actually, it goes a step further. The XR SDK providers see that their specific loader isn't running (because no XR SDK loader is running) and don't run themselves. So we end up with no truly-running provider!